db_upgrade_scripts/mosip_ida/sql/1.2.1.0_to_1.3.0_upgrade.sql

```diff
@@ -14,9 +14,13 @@ ALTER TABLE BATCH_JOB_EXECUTION_PARAMS ALTER COLUMN STRING_VAL TYPE VARCHAR(2500
 ALTER TABLE BATCH_JOB_EXECUTION_PARAMS RENAME STRING_VAL TO PARAMETER_VALUE;
 ALTER TABLE BATCH_JOB_EXECUTION DROP COLUMN JOB_CONFIGURATION_LOCATION;
 
-CREATE SEQUENCE BATCH_STEP_EXECUTION_SEQ START WITH 0 MINVALUE 0 MAXVALUE 9223372036854775807 NO CYCLE;
-CREATE SEQUENCE BATCH_JOB_EXECUTION_SEQ START WITH 0 MINVALUE 0 MAXVALUE 9223372036854775807 NO CYCLE;
-CREATE SEQUENCE BATCH_JOB_SEQ START WITH 0 MINVALUE 0 MAXVALUE 9223372036854775807 NO CYCLE;
+CREATE INDEX IF NOT EXISTS idx_job_name ON BATCH_JOB_INSTANCE(JOB_NAME);
+CREATE INDEX IF NOT EXISTS idx_job_key ON BATCH_JOB_INSTANCE(JOB_KEY);
 
-CREATE INDEX idx_job_name ON BATCH_JOB_INSTANCE(JOB_NAME);
-CREATE INDEX idx_job_key ON BATCH_JOB_INSTANCE(JOB_KEY);
\ No newline at end of file
+CREATE SEQUENCE IF NOT EXISTS BATCH_STEP_EXECUTION_SEQ START WITH 0 MINVALUE 0 MAXVALUE 9223372036854775807 NO CYCLE;
+CREATE SEQUENCE IF NOT EXISTS BATCH_JOB_EXECUTION_SEQ START WITH 0 MINVALUE 0 MAXVALUE 9223372036854775807 NO CYCLE;
+CREATE SEQUENCE IF NOT EXISTS BATCH_JOB_SEQ START WITH 0 MINVALUE 0 MAXVALUE 9223372036854775807 NO CYCLE;
+
+GRANT usage, SELECT ON ALL SEQUENCES
+   IN SCHEMA ida
+   TO idauser;
```

Comment:
1. Do we need to create the sequence or it already exists. If it is required then we should have rollback for the same.
2. Grant permission is required or not?

Reply:
https://github.com/mosip/id-authentication/blob/a43cfc7310562db7881838e4ec2c9c4ec0898d7e/db_scripts/mosip_ida/ddl/ida-fk.sql create sequence already present

Reply:
yes grant permission required